flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -191,7 +191,17 @@ object ScalarOperators {
           )
         )
     }
-  }  
+  }
+
+  def generateDistinctFrom(
+      nullCheck: Boolean,
+      left: GeneratedExpression,
+      right: GeneratedExpression)
+    : GeneratedExpression = {
+    val newleft = left.copy(nullTerm = GeneratedExpression.NEVER_NULL)
```

Comment:
This looks not correct to me. By setting the expression to never null, you basically compare the default values of expressions. Long has `0L` for example. For example, a `0 IS NOT DISTINCT FROM NULL` might return `true`.

Reply:
@twalthr I thought the similar way ago and met a problem. If we check left is null then need to generate left code first. However in generateEqual (generateOperatorIfNotNull) it also generate left code, such as 
```
${left.code}
${right.code}
if (${left.nullTerm}) {
  return ${right.nullTerm}
} else if (${right.nullTerm}) {
  return ${left.nullTerm}
} else {
  return generateEqual(left, right)
 (here will also generate ${left.code} and ${right.code} in generateOperatorIfNotNull)
}
```
I don't know how to make left and right code generate once if don't modify generateOperatorIfNotNull..

Reply:
It looks that there is one way which generateEqual first such as 
```
generateEquals(left, right)
check left.nullTerm and right.nullTerm
reset nullTerm=false and resultTerm
```
Do you think it is reasonable? Thanks~ @twalthr

Reply:
@xueyumusic You can also do it like this (pseudo code):
```
${left.code}
${right.code}
resultTerm;
if (${left.nullTerm}) {
  resultTerm = ${right.nullTerm}
} else if (${right.nullTerm}) {
  resultTerm = ${left.nullTerm}
} else {
  resultTerm = generateEqual(
    false, // no null check required in equals
    ${left.copy(code = NO_CODE)},
    ${right.copy(code = NO_CODE)})
}
```

Reply:
Thank you very much, @twalthr , This is an elegant solution. I updated the code